src/compiler/checker.ts

```diff
@@ -936,6 +937,31 @@ namespace ts {
             return false;
         }
 
+
+        function checkAndReportErrorForExtendingInterface(errorLocation: Node): boolean {
+            let parentClassExpression = errorLocation;
+            while (parentClassExpression) {

```

Comment:
`getAncestor` cannot be used here because it will confuse with `class A extends generateClass(Interface) {}`.
See the test case `classExtendsInterfaceInExpression`